agent/command/s3_copy.go

```diff
@@ -78,102 +98,132 @@ type s3Loc struct {
+		WeaklyTypedInput: true,
+		Result:           c,
+	})
+	if err != nil {
+		return errors.WithStack(err)
 	}
-	if err := c.validateParams(); err != nil {
-		return errors.Wrapf(err, "error validating %v params", c.Name())
+
+	if err := decoder.Decode(params); err != nil {
+		return errors.Wrapf(err, "error decoding %s params", c.Name())
 	}
-	return nil
+
+	return c.validate()
 }
 
 // validateParams is a helper function that ensures all
 // the fields necessary for carrying out an S3 copy operation are present
-func (c *s3copy) validateParams() error {
+func (c *s3copy) validate() error {
+	catcher := grip.NewSimpleCatcher()
+
+	// make sure the command params are valid
 	if c.AwsKey == "" {
-		return errors.New("s3 AWS key cannot be blank")
+		catcher.New("aws_key cannot be blank")
 	}
 	if c.AwsSecret == "" {
-		return errors.New("s3 AWS secret cannot be blank")
+		catcher.New("aws_secret cannot be blank")
 	}
+
 	for _, s3CopyFile := range c.S3CopyFiles {
-		if s3CopyFile.Source.Region == "" {
-			s3CopyFile.Source.Region = endpoints.UsEast1RegionID
-		}
-		if s3CopyFile.Destination.Region == "" {
-			s3CopyFile.Destination.Region = endpoints.UsEast1RegionID
-		}
-		if s3CopyFile.Source.Bucket == "" {
-			return errors.New("s3 source bucket cannot be blank")
-		}
-		if s3CopyFile.Destination.Bucket == "" {
-			return errors.New("s3 destination bucket cannot be blank")
-		}
 		if s3CopyFile.Source.Path == "" {
-			return errors.New("s3 source path cannot be blank")
+			catcher.New("s3 source path cannot be blank")
 		}
 		if s3CopyFile.Destination.Path == "" {
-			return errors.New("s3 destination path cannot be blank")
+			catcher.New("s3 destination path cannot be blank")
 		}
 		if s3CopyFile.Permissions == "" {
 			s3CopyFile.Permissions = s3.BucketCannedACLPublicRead
 		}
-
-		err := validateS3BucketName(s3CopyFile.Source.Bucket)
-		if err != nil {
-			return errors.Wrapf(err, "source bucket '%v' is invalid",
-				s3CopyFile.Source.Bucket)
+		if s3CopyFile.Source.Region == "" {
+			s3CopyFile.Source.Region = endpoints.UsEast1RegionID
 		}
-
-		err = validateS3BucketName(s3CopyFile.Destination.Bucket)
-		if err != nil {
-			return errors.Wrapf(err, "destination bucket '%v' is invalid",
-				s3CopyFile.Destination.Bucket)
+		if s3CopyFile.Destination.Region == "" {
+			s3CopyFile.Destination.Region = endpoints.UsEast1RegionID
+		}
+		// make sure both buckets are valid
+		if err := validateS3BucketName(s3CopyFile.Source.Bucket); err != nil {
+			catcher.Wrapf(err, "source bucket '%v' is invalid", s3CopyFile.Source.Bucket)
+		}
+		if err := validateS3BucketName(s3CopyFile.Destination.Bucket); err != nil {
+			catcher.Wrapf(err, "destination bucket '%v' is invalid", s3CopyFile.Destination.Bucket)
 		}
+
 	}
 
-	return nil
+	return catcher.Resolve()
 }
 
-// Execute carries out the s3copy command - this is required
-// to satisfy the 'Command' interface
+// Implementation of Execute.  Expands the parameters, and then copies the
```

Comment:
Can this still say `Execute expands the parameters, ...`? Go documentation comments conventionally start with the name of the thing being documented (i.e. `Execute`).